### DISCUSSION TITLE
After updating to React 18, forms using RHF don't have the proper data in the submit handler

### DISCUSSION THREAD

**Discussion by billsaysthis** [react-hook-form/react-hook-form#10664] [Q&A]
One of the fields is being passed as undefined despite a value being set in the form itself. Generally (always possibly) the field having the problem is a Select. This has been working correctly for 2+ years with React 16.x but after updating to React 18 today we're seeing this wherever RHF is used with a select.

No JS error or warning in the console. Using version 7.45.1 of RHF and React 18.2.0.

**Comment by leapful** [comment]
@billsaysthis 

Please provide a codesandbox url to re-produce the issue.
